pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -832,3 +834,20 @@ func (m *ManagerImpl) GetDevices(podUID, containerName string) []*podresourcesap
 	defer m.mutex.Unlock()
 	return m.podDevices.getContainerDevices(podUID, containerName)
 }
+
+// ShouldResetExtendedResourceCapacity returns whether the extended resources should be zeroed or not,
+// depending on whether the node has been recreated. Absence of the checkpoint file strongly indicates the node
+// has been recreated.
+func (m *ManagerImpl) ShouldResetExtendedResourceCapacity() bool {
+	if utilfeature.DefaultFeatureGate.Enabled(features.DevicePlugins) {
+		// return false if checkpoint exists
+		m.mutex.Lock()
+		checkpoints, err := m.checkpointManager.ListCheckpoints()
+		m.mutex.Unlock()
+		if err != nil {
```

Comment:
In case checkpoint file doesn't exist, would ListCheckpoints() return error or zero-length checkpoints?

Reply:
ListCheckpoints() will return zero-length checkpoints. When calling checkpointManager.List(), the directory will already haven been created during initialization: https://github.com/kubernetes/kubernetes/blob/7f23a743e8c23ac6489340bbb34fa6f1d392db9d/pkg/kubelet/util/store/filestore.go#L44.